auditbeat/module/auditd/audit_linux.go

```diff
@@ -217,11 +231,15 @@ func (ms *MetricSet) initClient() error {
 			return errors.Wrap(err, "failed to enable auditing in the kernel")
 		}
 	}
-
-	if err := ms.client.SetPID(libaudit.NoWait); err != nil {
-		return errors.Wrap(err, "failed to set audit PID")
+	if err := ms.client.WaitForPendingACKs(); err != nil {
```

Comment:
Thought of one more thing. 😄
If the audit config is locked, will any of the above config changes result in a failure?

Reply:
Yes, currently this is failing either in this line or the SetPID below.

I think we need to handle configuration locking differently as it will only work with a multicast socket. See [this comment](https://github.com/elastic/beats/issues/6019#issuecomment-356524906) I left on the issue.